pytest_client_tools/plugin.py

```diff
@@ -256,3 +263,12 @@ def pytest_runtest_logfinish(nodeid, location):
     if node_running_data.logfile.exists():
         node_running_data.artifacts.copy(node_running_data.logfile)
     logging.getLogger().handlers.remove(node_running_data.handler)
+
+
+def pytest_runtest_setup(item):
+    envnames = [mark.args[0] for mark in item.iter_markers(name="env")]
+    the_environment = _settings.get("env_for_dynaconf") or "development"
+    if envnames and the_environment not in envnames:
+        pytest.skip(
+            f"test requires a dynaconf environment to be one of those: {envnames}",
+        )
```

Comment:
I think that, rather than implementing it here, it can be implemented as proper marker; something like this:
```python
def pytest_client_tools_env(*envnames):
    return pytest.mark.skipif(
        the_current_environment not in envnames,
        reason="different environment"
    )
```

Reply:
I do not understand how to write this decorator. I've copied the marker env from https://docs.pytest.org/en/7.1.x/example/markers.html#custom-marker-and-command-line-option-to-control-test-runs They sudgested to use explicit markers creation since pytest --list-markers will list them.   Is it OK for a code to remain as it is? https://docs.pytest.org/en/7.1.x/example/markers.html#registering-markers  I can move some specific logic out of the plugin.py to config.py.

Reply:
> I do not understand how to write this decorator.

Not sure what do you mean "how to write": what I suggested is a working way, and only needs to get a proper logic/condition to pass to `skipif()`. Using it is no different than any other decorator/marker:

```python
@pytest_client_tools_env("env1")
def test_that_runs_only_on_env1():
```
It needs to still be added to the markers manually, in the same way of the currently implemented `env()`.